src/aws/flb_aws_util.c

```diff
@@ -927,3 +932,62 @@ flb_sds_t flb_get_s3_key(const char *format, time_t time, const char *tag,
         }
         return NULL;
 }
+
+/*
+ * Replace %f and %L in time_key_format with milliseconds and nanoseconds. And get the
+ * time format output and output length with strftime().
+ */
+size_t create_time_formatter(char **out_buf_ptr, const char *time_key_format,
+                             struct flb_time *tms)
+{
+    char millisecond_string[FLB_AWS_MILLISECOND_FORMATTER_LENGTH+1];
+    char nanosecond_string[FLB_AWS_NANOSECOND_FORMATTER_LENGTH+1];
```

Comment:
to follow your convention, please rename to `millisecond_str` and `nanosecond_str`